### PR TITLE
PKCS #11 PAL MISRA Fix

### PR DESCRIPTION
**Pull request by lundinc2**
PKCS #11 PAL MISRA Update

Description
-----------
This commit fixes some of the MISRA errors caused by the PKCS #11 PAL API. It makes most functions use the PKCS #11 data types, as well as reduces the amount of casting being done in the PKCS #11 stack by as much as possible.

Checklist:
----------

- [x ] I have tested my changes. No regression in existing tests.
- [ x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c [hunk 1]
```diff
@@ -1025,58 +1026,57 @@ static CK_RV prvDeleteObjectFromList( CK_OBJECT_HANDLE xAppHandle )
...
     /* coverity[misra_c_2012_rule_10_5_violation] */
     CK_BBOOL xObjectFound = ( CK_BBOOL ) CK_FALSE;
-    int32_t lInsertIndex = -1;
-    int32_t lSearchIndex = pkcs11configMAX_NUM_OBJECTS - 1;
-
-    xGotSemaphore = xSemaphoreTake( xP11Context.xObjectList.xMutex, portMAX_DELAY );
+    uint32_t ulSearchIndex = 0;
 
-    if( xGotSemaphore == pdTRUE )
+    if( pdTRUE == xSemaphoreTake( xP11Context.xObjectList.xMutex, portMAX_DELAY ) )
```
Comment (aggarg) [inline libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:1039]: What is the reason for this change? I think it is easier to debug if the return value is stored in a local variable.

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c [hunk 2]
```diff
@@ -2856,7 +2872,10 @@ CK_DECLARE_FUNCTION( CK_RV, C_FindObjectsInit )( CK_SESSION_HANDLE hSession,
     /* Malloc space to save template information. */
     if( xResult == CKR_OK )
     {
-        pxFindObjectLabel = pvPortMalloc( pTemplate->ulValueLen + 1UL ); /* Add 1 to guarantee null termination for PAL. */
+        /* Add a NULL terminator. */
```
Comment (aggarg) [inline libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:2875]: You mean add space for a NULL terminator.

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c (hunk 1 above)
Comment (lundinc2) [reply libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:1039]: I like the conciseness, and not having to declare a variable. I don't mind reverting this

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c (hunk 2 above)
Comment (lundinc2) [reply libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:2875]: Yes, I'll add a comment to the memset as well, as that is what is guaranteeing the memory at the end is NULL
